controllers/scopeinstance_controller.go

```diff
@@ -104,210 +131,167 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
-			existingCRB.Subjects = crb.Subjects
-
-			// server-side apply patch
-			existingCRB.ManagedFields = nil
-			if err := r.Client.Patch(ctx,
-				existingCRB,
-				client.Apply,
-				client.FieldOwner(siCtrlFieldOwner),
-				client.ForceOwnership); err != nil {
-				return err
-			}
+	if len(rbList.Items) > 1 {
+		return fmt.Errorf("more than one RoleBinding found for ClusterRole %s", cr.GenerateName)
+	}
 
+	// GenerateName is immutable, so create the object if it has changed
+	if len(rbList.Items) == 0 {
+		if err := r.Client.Create(ctx, rb); err != nil {
+			return err
 		}
-	} else {
-		// it will iterate over the namespace and createrole bindings for each cluster roles
-		for _, namespace := range in.Spec.Namespaces {
-			for _, cr := range st.Spec.ClusterRoles {
-				rb := &rbacv1.RoleBinding{
-					ObjectMeta: metav1.ObjectMeta{
-						GenerateName: cr.GenerateName + "-",
-						Namespace:    namespace,
-						Labels: map[string]string{
-							scopeInstanceUIDKey:           string(in.GetUID()),
-							scopeTemplateUIDKey:           string(st.GetUID()),
-							scopeInstanceHashKey:          util.HashObject(in.Spec),
-							scopeTemplateHashKey:          util.HashObject(st.Spec),
-							clusterRoleBindingGenerateKey: cr.GenerateName,
-						},
-						OwnerReferences: []metav1.OwnerReference{{
-							APIVersion: in.APIVersion,
-							Kind:       in.Kind,
-							Name:       in.GetObjectMeta().GetName(),
-							UID:        in.GetObjectMeta().GetUID(),
-						}},
-					},
-					Subjects: cr.Subjects,
-					RoleRef: rbacv1.RoleRef{
-						Kind:     "ClusterRole",
-						Name:     cr.GenerateName,
-						APIGroup: "rbac.authorization.k8s.io",
-					},
-				}
-
-				rbList := &rbacv1.RoleBindingList{}
-				if err := r.Client.List(ctx, rbList, &client.ListOptions{
-					Namespace: namespace,
-				}, client.MatchingLabels{
-					scopeInstanceUIDKey:           string(in.GetUID()),
-					scopeTemplateUIDKey:           string(st.GetUID()),
-					clusterRoleBindingGenerateKey: cr.GenerateName,
-				}); err != nil {
-					return err
-				}
+		return nil
+	}
 
-				if len(rbList.Items) > 1 {
-					return fmt.Errorf("more than one roleBinding found for ClusterRole %s", cr.GenerateName)
-				}
+	log.Log.Info("Updating existing rb", "namespaced", rbList.Items[0].GetNamespace(), "name", rbList.Items[0].GetName())
 
-				// GenerateName is immutable, so create the object if it has changed
-				if len(rbList.Items) == 0 {
-					if err := r.Client.Create(ctx, rb); err != nil {
-						return err
-					}
-					continue
-				}
+	existingRB := &rbList.Items[0]
 
-				log.Log.Info("Updating existing rb", "namespaced", rbList.Items[0].GetNamespace(), "name", rbList.Items[0].GetName())
+	if util.IsOwnedByLabel(existingRB.DeepCopy(), in) &&
+		reflect.DeepEqual(existingRB.Subjects, rb.Subjects) &&
+		reflect.DeepEqual(existingRB.Labels, rb.Labels) {
+		log.Log.Info("existing RoleBinding does not need to be updated")
```

Comment:
Same as for ClusterRoleBinding, we should print out some sort of ID or name. If I'm looking at the logs I'm there for a reason so let's give them the information they might want to see.

Reply:
updated in commit https://github.com/operator-framework/oria-operator/pull/48/commits/77b7c6f630462f260b85e7e8d3c753944298c53b